### ISSUE TITLE
keycloak running on kubernetes | not able to browse admin console  giving (blocked mixed content)

### ISSUE THREAD

**Issue by mootezbessifi** [keycloak/keycloak#13865]
### Describe the bug

I am running keycloak 19.0.1 on kubernetes 1.21.4
I have one internal nginx ingress controller installed on kubernetes and not exposed to outside, and i have another nginx server which is in the DMZ exposed to external and running as reverse proxy.

1. I used to un keycloak with the option `PROXY_ADDRESS_FORWARDING=true` from inside kubernetes
2. I used to configure the below Ingress file for it to be exposed through the nginx ingress controller:

``` 
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
  labels:
    app: keycloak
  name: keycloak
  namespace: fms-preprod
spec:
  rules:
  - host: keycloak-test-ssl.stcs.com.sa
    http:
      paths:
      - backend:
          service:
            name: keycloak
            port:
              number: 8443
        path: /
        pathType: Prefix
```

3. I used to define the below configuration inside the external nginx reverse proxy:

```
server {
    listen 80 default_server;
    server_name _;
    return 301 https://$host$request_uri;
}

server {

  listen 443 ssl;
  server_name keycloak-test-ssl.stcs.com.sa ;
  location / {

        proxy_pass http://ingress;
	proxy_set_header X-Forwarded-For $host;
	proxy_set_header X-Forwarded-Proto $scheme;
	proxy_set_header Upgrade $http_upgrade;
	proxy_set_header Connection 'upgrade';
	proxy_set_header Host $host;
        client_max_body_size 100M;

}
    ssl_certificate       /etc/ssl/blabla/bundle1.crt;
    ssl_certificate_key   /etc/ssl/blabla/bundle1.key;
}
```

Everything is working fine using the http without SSL configuration and i am able to browse the admin console, but once configuring the SSL on the nginx external reverse proxy as mentioned above, the first page of keycloak is loading fine but the admin console is blocked loading and giving the error as mentioned in the below pic.

<img width="1787" alt="Screen Shot 2022-08-18 at 5 53 44 PM" src="https://user-images.githubusercontent.com/5040743/185427064-aba8b752-f2b3-4e74-851e-99a93ddf2d27.png">

Any one tell me what is wrong or missing in my configuration would be helpful for me
thanks in advance 

### Version

19.0.1

### Expected behavior

keycloak has to work properly behind nginx Reverse proxy with SSL

### Actual behavior

keycloak failed loading admin console giving `blocked mixed content`

### How to Reproduce?

details given in above description

**Comment by tyokyo320** [comment]
I also encountered the same problem. But I solved the problem after adding the following settings.

```yaml
- name: KC_PROXY
  value: "edge"
```

**Comment by mootezbessifi** [comment]
@tyokyo320  this is the definition of my keycloak deployment (twhere KC_PROXY is there ) and still not working 

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: keycloak
  namespace: fms-preprod
  labels:
    app: keycloak
    component: keycloak
spec:
  replicas: 1
  selector:
    matchLabels:
      app: keycloak
      component: keycloak
  template:
    metadata:
      labels:
        app: keycloak
        component: keycloak
    spec:
      containers:
      - name: keycloak
        image: quay.io/keycloak/keycloak:19.0.1
        args: ["start-dev"]
        env:
        - name: KEYCLOAK_ADMIN
          value: "admin"
        - name: KEYCLOAK_ADMIN_PASSWORD
          value: "xxxxxxxxxxx"
        - name: KC_PROXY
          value: "edge"
        - name: PROXY_ADDRESS_FORWARDING
          value: "true"
        ports:
        - name: http
          containerPort: 8080
        readinessProbe:
          httpGet:
            path: /realms/master
            port: 8080

```

**Comment by pedroigor** [comment]
@mootezbessifi What is the URL you are trying to access the administration console? Is it using the same host your ingress is exposing?

**Comment by mootezbessifi** [comment]
@pedroigor yes indeed. As I said I was able to access the home page but stuck on browsing the admin console. Getting the above screenshot

**Comment by pedroigor** [comment]
@mootezbessifi I can reproduce the behavior. You should either forward the headers (e.g.: `X-Forwarded-*`) in your ingress, similarly as you are doing in the external proxy, or set the `hostname-url` and `hostname-admin-url` options.

However, the `hostname-url` and `hostname-admin-url` are not available from `19.0.1` but `19.0.2+`.

**Comment by mootezbessifi** [comment]
@pedroigor  could you tell me which headers i need to farward and how to forward those headers using ingress object??

**Comment by pedroigor** [comment]
@mootezbessifi I'm not sure. But once `19.0.2` is released you should be able to force a base URL to that host, port, scheme, are all inferred from this base URL.

**Comment by mootezbessifi** [comment]
@pedroigor could you please share a doc link for how to enforce this ?

**Comment by pedroigor** [comment]
The updated doc is here https://github.com/keycloak/keycloak/blob/main/docs/guides/src/main/server/hostname.adoc. Should be available once `19.0.2` is released.
